### PR TITLE
Add migration steps for `TemplatedConfigLoader` to `OmegaConfigLoader`

### PR DESCRIPTION
**Pull request by merelcht**
## Description
Second part of https://github.com/kedro-org/kedro/issues/2699

## Development notes
Left to do: 

- [x] Add section on globals. Blocked by https://github.com/kedro-org/kedro/issues/2794
- [x] Update `0.18.X` to version that includes globals for `OmegaConfigLoader`, probably `0.18.13`.
- [x] Link to section about globals when https://github.com/kedro-org/kedro/issues/2794 is merged

## Questions for reviewers:
- [ ] There is some overlap in steps between the migration from `ConfigLoader` to `OmegaConfigLoader` and `TemplatedConfigLoader` to `OmegaConfigLoader`. I just repeated them, because I think users will just skip the first section if they're only interested in `TemplatedConfigLoader`. Does that seem right?
- [ ] Does the diff highlighting look okay in the examples?

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [inline docs/source/configuration/config_loader_migration.md:209]: It's not clear yet if this works, so might need to remove.

File: docs/source/configuration/config_loader_migration.md
Comment (datajoely) [inline docs/source/configuration/config_loader_migration.md:163]: you may want to define this as it currently doesn't have context

File: docs/source/configuration/config_loader_migration.md
Comment (datajoely) [inline docs/source/configuration/config_loader_migration.md:199]:
```suggestion
-   type: "${datasets.spark}"
```
Slightly more legible and you don't really need to reiterate this

File: docs/source/configuration/config_loader_migration.md
Comment (datajoely) [reply docs/source/configuration/config_loader_migration.md:209]: can you escape the `.` in `.csv`?

File: docs/source/configuration/config_loader_migration.md
Comment (datajoely) [inline docs/source/configuration/config_loader_migration.md:212]:
```suggestion
### 8. Deprecation of Jinja2
```

File: docs/source/configuration/config_loader_migration.md
Comment (datajoely) [inline docs/source/configuration/config_loader_migration.md:214]: I wonder if we want to provide some opinion here on why Jinja is a suboptimal solution here, not built for a whitespaced language etc.

Or point to articles like [this one](https://sre.google/workbook/configuration-specifics/) (last paragraph of "Pitfall 1")

**Review by datajoely — COMMENTED** [review]
Added some thoughts, hopefully helpful!

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [reply docs/source/configuration/config_loader_migration.md:72]: Good point. I'll wait until the globals functionality has been merged to set the lower bound here.

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [inline docs/source/configuration/config_loader_migration.md:66]:
```suggestion
### 1. Install the required library
```

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [inline docs/source/configuration/config_loader_migration.md:91]:
```suggestion
### 3. Import statements
```

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [inline docs/source/configuration/config_loader_migration.md:100]:
```suggestion
### 4. File format support
```

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [inline docs/source/configuration/config_loader_migration.md:103]:
```suggestion
### 5. Load configuration
```

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [inline docs/source/configuration/config_loader_migration.md:119]:
```suggestion
### 6. Templating of values
```

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [inline docs/source/configuration/config_loader_migration.md:234]:
```suggestion
### 9. Exception handling
```

**Comment by merelcht** [comment]
@stichbury I tried resolving some of the Vale comments, but I wasn't too succesful 😅 Would you mind reviewing?

**Comment by astrojuanlu** [comment]
@stichbury is on PTO, I suspect the bogus "sentence-style capitalization" warnings come because of the numbers 🤔 Other than that, you're capitalizing only the first word of the header, which as far as I know is the intended style. Maybe we can ignore those for now?

File: docs/source/configuration/config_loader_migration.md
Comment (ankatiyar) [inline docs/source/configuration/config_loader_migration.md:119]: I think, under this section, it would also be helpful to flag that the variable interpolation is only scoped to a particular configuration and the same environment. For eg., templated keys from `local` will not overwrite `base` and the resolution of the keys happens within the environment which is different from how `TemplatedConfigLoader` uses the globals.

Comment (noklam) [reply docs/source/configuration/config_loader_migration.md:119]: Second on this. Do we already have the relevant docs in OmegaConfigLoader's page? Obviously just move everything to `global` would work(this is how template config loader works), but we may want to advise to use the `catalog_globals.yml` etc to replace the use of `globals.py` for certain things.

File: docs/source/configuration/config_loader_migration.md
Comment (noklam) [inline docs/source/configuration/config_loader_migration.md:189]: Should we add a link in case they name it something other than `globals.yml`?

Comment (noklam) [reply docs/source/configuration/config_loader_migration.md:189]: What's the replacement for `globals_dict`?

**Review by noklam — COMMENTED** [review]
I had some quick comments, I want to have a second look on the rendered version but the RTD is failing.

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [reply docs/source/configuration/config_loader_migration.md:189]: There's no replacement. That's why I wrote "The `OmegaConfigLoader` requires global values to be provided in a `globals.yml` file.", but I can explicitly mention that the `globals_dict` is not supported with OCL for clarity.

Comment (merelcht) [reply docs/source/configuration/config_loader_migration.md:189]: "Should we add a link in case they name it something other than globals.yml?" @noklam what should we link to?

Comment (noklam) [reply docs/source/configuration/config_loader_migration.md:190]: @merelcht How about https://docs.kedro.org/en/latest/configuration/advanced_configuration.html#how-to-use-global-variables-with-the-omegaconfigloader?

Btw I think we should bubble up OmegaConfigLoader docs since we are trying to make it default. The hierarchy is also broken there is nothing inside OmegaConfigLoader. Cc @stichbury @ankatiyar 
image

**Comment by merelcht** [comment]
> I had some quick comments, I want to have a second look on the rendered version but the RTD is failing.

The RTD is passing again 🙂

File: docs/source/configuration/config_loader_migration.md
Comment (merelcht) [reply docs/source/configuration/config_loader_migration.md:190]: > The hierarchy is also broken there is nothing inside OmegaConfigLoader.

I don't think it's broken, we don't have any sub-sections under `OmegaConfigLoader` because those are all under "Advanced Kedro configuration". We should probably clean this up when all docs on `ConfigLoader` and `TemplatedConfigLoader` are removed as part of: https://github.com/kedro-org/kedro/issues/2692

Comment (ankatiyar) [reply docs/source/configuration/config_loader_migration.md:190]: @noklam We will be doing some of the moving around sections in the docs for https://github.com/kedro-org/kedro/issues/2900 after the "make starters use `OmegaConfigLoader` by default" PRs are merged. (@lrcouto)

**Comment by ankatiyar** [comment]
Unrelated but I see that the Vale reviewer action keeps doing a whole round of reviews after every push. I'll try to see if there's a way for it to be less annoying 😅

File: docs/source/configuration/config_loader_migration.md
Comment (astrojuanlu) [inline docs/source/configuration/config_loader_migration.md:72]: This requires quotes!

```suggestion
pip install "kedro>=0.18.13, <0.19.0"
```

File: docs/source/configuration/config_loader_migration.md
Comment (astrojuanlu) [inline docs/source/configuration/config_loader_migration.md:97]: nit: Maybe remove the empty line here?

```suggestion
- from kedro.config import TemplatedConfigLoader
+ from kedro.config import OmegaConfigLoader
```

**Review by astrojuanlu — APPROVED** [review]
This is fantastic ⭐ Thanks a lot @merelcht and reviewers! Left a couple of comments, other than that let's get this merged :shipit:

**Pull request merged**
